### PR TITLE
feature/#154 refactor global config checker

### PR DESCRIPTION
**Pull request by toan-quach**
refactor repo type of _ACCEPTED_REPOSITORY_TYPES in global config checker for expansion

### CODE REVIEW

File: src/taipy/config/checker/_checkers/_gLobal_config_checker.py [hunk 1]
```diff
@@ -19,7 +21,7 @@
 
 class _GlobalConfigChecker(_ConfigChecker):
     
-    _ACCEPTED_REPOSITORY_TYPES = ["filesystem", "sql"]
+    _ACCEPTED_REPOSITORY_TYPES: Set[str] = {"filesystem"}
```
Comment (trgiangdo) [inline src/taipy/config/checker/_checkers/_gLobal_config_checker.py:24]: I thought we were going to leave this an empty set?

File: tests/config/checker/checkers/test_gLobal_config_checker.py [hunk 2]
```diff
@@ -19,14 +19,17 @@
...
+    def test_check_repository_type_mock_value_for_expansion(self):
         config = _Config()
-        config._global_config.repository_type = "sql"
+        config._global_config.clean_entities_enabled = False
+        config._global_config.repository_type = "mock_repo_type"
         collector = IssueCollector()
         _GlobalConfigChecker(config, collector)._check()
         assert len(collector.warnings) == 0
-
+
```
Comment (trgiangdo) [inline tests/config/checker/checkers/test_gLobal_config_checker.py]: Unnecessary tabs here I guess?

**Review by trgiangdo — DISMISSED** [review]
LGTM! Only have some minor comments

File: src/taipy/config/checker/_checkers/_gLobal_config_checker.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/config/checker/_checkers/_gLobal_config_checker.py:24]: yeah... but then the default value for repo type in the default GlobalAppConfig is "filesystem". So we need to have it so the config doesn't raise any warning, which makes sense to me either way. :D

Comment (trgiangdo) [reply src/taipy/config/checker/_checkers/_gLobal_config_checker.py:24]: ahh true 🤣

**Review by trgiangdo — APPROVED** [review]
Great

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**